core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -150,11 +148,7 @@ class RangePolicy : public Impl::PolicyTraits<Properties...> {
   template <class... Args>
   inline RangePolicy(const member_type work_begin, const member_type work_end,
                      Args... args)
-      : RangePolicy(typename traits::execution_space(), work_begin, work_end) {
-    check_bounds_validity();
-    set_auto_chunk_size();
-    set(args...);
```

Comment:
The `set(args...);` would not be executed with the change you propose.

Reply:
Fixed to call the constructor that takes in a parameter pack `args`.